harness/determined/core/_train.py

```diff
@@ -89,10 +112,55 @@ def report_training_metrics(
             data=det.util.json_encode(body),
         )
 
-        # Also sync tensorboard.
-        if self._tbd_writer and self._tbd_mgr:
+        if self._tbd_sync_mode == TensorboardSyncMode.Auto:
+            self.write_training_tensorboard_metrics(steps_completed, metrics, batch_metrics)
+            if self._tbd_mgr:
+                self._tbd_mgr.sync()
+
+    def write_training_tensorboard_metrics(
+        self,
+        steps_completed: int,
+        metrics: Dict[str, Any],
+        batch_metrics: Optional[List[Dict[str, Any]]] = None,
+    ) -> None:
+        if self._tbd_writer:
             self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
-            self._tbd_mgr.sync()
+
+    def get_tensorboard_base_path(self) -> Optional[pathlib.Path]:
```

Comment:
take the word `base` out of the name, since it doesn't really add any meaning.

Also, is there any way we can make sure this returns a Path instead of an Optional[Path]?  It seems like it would be very inconvenient to callers for this to sometimes return None.